plugins/scaffolder-backend/src/ScaffolderPlugin.test.ts

```diff
@@ -0,0 +1,1004 @@
+/*
+ * Copyright 2025 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { IncomingMessage } from 'http';
+import request from 'supertest';
+import waitForExpect from 'wait-for-expect';
+
+import { createBackendModule } from '@backstage/backend-plugin-api';
+import { startTestBackend } from '@backstage/backend-test-utils';
+import { CatalogClient } from '@backstage/catalog-client';
+import { stringifyEntityRef } from '@backstage/catalog-model';
+import { TemplateEntityV1beta3 } from '@backstage/plugin-scaffolder-common';
+import { scaffolderAutocompleteExtensionPoint } from '@backstage/plugin-scaffolder-node/alpha';
+
+import { scaffolderPlugin } from './ScaffolderPlugin';
+
+jest.mock('@backstage/catalog-client');
+
+describe('scaffolderPlugin', () => {
+  const getEntityByRefSpy = jest.spyOn(
```

Comment:
Wanna quickly jump in and highlight this option, which is a bit simpler to maintain: https://github.com/backstage/backstage/blob/5108226057c62aa0a66dc9f8bb0b7a9ebb258a9c/packages/cli/templates/backend-plugin/src/plugin.test.ts.hbs#L53

Reply:
Thanks, Patrik. I've incorporated the change in its own commit since I ended touching more files than I anticipated, and wasn't sure if we're okay with that.